### ISSUE TITLE
Write PyCuda version of Encode file

### ISSUE THREAD

**Issue by zenlc2000** [zenlc2000/pycuda-euler#4]
Module file to encapsulate the Encode GPU kernel code.

**Comment by zenlc2000** [comment]
Pushed the first draft of pyencode.py. Was able to do some manual testing & debugging on the first function: encode_lmer_device but I have a shared memory problem (I think) causing problems. I was able to nudge it around a bit so now it looks like a synchronization error:

  File "/Users/zen/Code/git/pycuda-euler/src/pyencode.py", line 127, in encode_lmer_device
    encode_lmer(drv.In(buffer), drv.Out(d_lmers), gpu_lmerLength, block=block, grid=grid, shared=readLength+31)
  File "/anaconda/envs/Python3/lib/python3.5/site-packages/pycuda-2016.1-py3.5-macosx-10.5-x86_64.egg/pycuda/driver.py", line 405, in function_call
    Context.synchronize()
pycuda._driver.LaunchError: cuCtxSynchronize failed: unspecified launch failure
We've got an error while stopping in post-mortem: <class 'KeyboardInterrupt'>

PyCUDA WARNING: a clean-up operation failed (dead context maybe?)
cuMemFree failed: unspecified launch failure

With class prep & teaching I don't know if I'll get back to this until way late tonight.
